app/javascript/application.js

```diff
@@ -28,4 +28,19 @@ document.addEventListener("turbo:load", async () => {
   // Start argo after free-jqgrid has been loaded
   new Argo().initialize()
 })
+
+window.addEventListener('turbo:before-cache', function () {
+  // Close any lingering open modal windows
+  // $('.modal:visible').modal('hide');
+  // $('.modal-backdrop').remove();
+  if (document.body.classList.contains('modal-open')) {
+    $('.modal')
```

Comment:
Please avoid using jquery.

Reply:
Is there a reason you aren't using the bootstrap modal API?

Reply:
At this point, I'm at the "adapt what was in the branch before by adding copypasta" phase. It's complicated by not being obvious how to reproduce this locally. I'll check the BS4 docs.

Reply:
But I'm wondering if we can add more `data-dismiss="modal"` to modal buttons that submit instead?

Reply:
@jcoyne I'll be pushing stuff to this branch here and there. feel free to disregard until it's undrafted. I don't think it'll be fruitful to code review what's here until then.

Reply:
I would expect the only bit of code needed would be `$('.modal').modal('hide')`

Reply:
yeah, i think the overlay is the `$('.modal-backdrop').remove();`

Reply:
I'm finding it hard to reconcile two things: "don't use JQuery" and "use the Bootstrap 4 API". Looking at the BS4 docs, it appears their examples all use JQuery. 🤷🏻‍♂️